### ISSUE TITLE
ChoiceGroup A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6844]
Tracking all of the issues brought up with the 10/23 accessibility pass on ChoiceGroup  control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B79a468c4-6973-4d37-aa26-0e916c22648d%7D&action=default&uid=%7B79A468C4-6973-4D37-AA26-0E916C22648D%7D&ListItemId=1487&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [x] ~~Bug 1: [Accessibility][Usable]:  Screen starts scrolling down after navigating last 'Mark displayed items as read after' drop down items using down arrow key.~~ Doesn't repro anymore.
- [x] ~~Bug 2: [Accessibility][Keyboard navigation]: Focus order is not correct while navigating 'ChoiceGroup with dropdown' radio buttons.~~ Fixed in #8107.
- [x] ~~Bug 3: [Accessibility][Programmatic access]: 'Required' text is not visually present for 'Here is a custom label' mandatory field.~~ Fixed in #8118.
- [x] ~~Bug 4: [Accessibility][Programmatic access]: Name of the drop down under 'ChoiceGroup with dropdown' section is not provided.~~ Fixed in #8107.
- [x] ~~Bug 5: [Accessibility][Usable]: Narrator focus is not visible while navigating through radio buttons.~~ Fixed in #8126.
- [x] ~~Bug 6: [Accessibility][Usable]: Scan mode focus goes on some hidden content after 'Show code' button.~~ Doesn't repro anymore.
- [x] ~~Bug 7: [Accessibility][Screen reader]: Radio buttons have two scan mode focus.~~ By design. 
- [x] ~~Bug 8: [Accessibility][Screen reader]: Narrator does not convey collapsed state of 'Mark displayed items as read after' drop down.~~ Duplicate of #5752.
- [x] ~~Bug 9: [Accessibility][Programmatic access]: 'Selected state' is not defined for 'ChoiceGroup with icons- week radio button.~~ Fixed in #8112.
- [x] ~~Bug 10: [Accessibility][Programmatic access]: 'Role=application' should not be implemented.~~ By design.
- [x] ~~Bug 11: [Accessibility][Usable][Suggestion]: Radio group could be better labelled using 'field set' and 'legends'.~~ By design.

Component: ChoiceGroup
@micahgodbolt

**Comment by khmakoto** [comment]
**Bug 2**

Hi @TestAccessibilityGIT can you provide more detail on what do you mean by "focus order not being correct". What are your expectations for the focus order while navigating "ChoiceGroup with dropdown" radio buttons.

**Comment by khmakoto** [comment]
**Bug 3**

Hi @TestAccessibilityGIT, can you tell me what exactly do you mean by "'Required' text not visually present"?

**Comment by khmakoto** [comment]
**Bug 4**

Hi @TestAccessibilityGIT, just wanted to check if you meant that for the "label" and "aria-label" attributes in the dropdown were not set when you say "name of dropdown under 'ChoiceGroup with dropdown' section is not provided".

**Comment by khmakoto** [comment]
**Bugs 5 and 6**

Hi @TestAccessibilityGIT  can you please tell me which steps to follow to repro bugs 5 and 6?

**Comment by TestAccessibilityGIT** [comment]
**Bug 2**
Hi @khmakoto when we have selected "Option B" under "ChoiceGroup with dropdown" then on tabbing, focus should land on "Option B". But here focus is landing on "Mark displayed items as read after 5 seconds" which doesn't seems to be logical.

**Comment by TestAccessibilityGIT** [comment]
**Bug 3**
Hi @khmakoto, by "require/mandatory" field we mean "*" sign which generally displayed to show that it's mandatory/required to be field. "*" has been implemented for other choices but not in case of "Here is a custom label".

**Comment by TestAccessibilityGIT** [comment]
**Bug 4**

Hi @khmakoto, here screen reader is reading the "Pick one" label which is actually the label for group of radio groups. In this case screen reader should read the name as "Mark displayed items as read after" then the selected options.

**Comment by TestAccessibilityGIT** [comment]
**Bug 5 and 6**
Hi @khmakoto, please turn on screen reader in scan mode then follow the steps mentioned in bug report.
Please let us know if you still find any issue which reproducing this issue.

**Comment by khmakoto** [comment]
Crossing out __Bug 2__ as it seems that this issue has been fixed and doesn't repro anymore.

**Comment by khmakoto** [comment]
__Bug 3__

Hi @TestAccessibilityGIT, on the Sharepoint Sheet, in the 'Expected result' section of this bug, you say:

>'Required' attribute should not be implemented on each radio options. Screen reader shouId convey this information on selected option in forward navigation/backward navigation.

Would the fix here be to only implement the `required` attribute on the selected radio options and leave it out of the other ones?

**Comment by khmakoto** [comment]
Crossing out __Bug 8__ as it is an underlying issue with the `Dropdown` component that is being tracked in #5752.

**Comment by TestAccessibilityGIT** [comment]
+Amit

**Comment by khmakoto** [comment]
__Bug 3__

After going through the spec I've found out the recommendation for `required` is to have the prop set on all radio buttons of a group:

![image](https://user-images.githubusercontent.com/7798177/53457070-bb0d0f00-39e5-11e9-9306-0de1654ace02.png)

Given this, I'll just fix the first part of the issue.

**Comment by v-doku** [comment]
Bug3:

As a best screen reader experience and coding practice we should implement radio group as fieldset and legend as mentioned in Bug 11, Upon implementing this screen reader will read required only once if we provide aria-required=true attribute in legend/fieldset. Implementing Required in each radio button will confuse the user as he can select only one radio button at a time and also here for disabled radio button as well its being read as required which is confusing.

**Comment by khmakoto** [comment]
Crossing out **Bug 6** as it seems that this issue has been fixed and doesn't repro anymore.

**Comment by TestAccessibilityGIT** [comment]
+Amit
________________________________

**Comment by khmakoto** [comment]
Crossing out **Bug 7** after confirming with the narrator team that this is the behavior by design in scan mode.

**Comment by khmakoto** [comment]
Crossing out __Bug 11__ because it was a suggestion to improve screen reader readability related to the second part of __Bug 3__, which, after conversations with the narrator team, has been classified as behavior that is expected by design.

**Comment by khmakoto** [comment]
All bugs have been taken care of, so I'll be closing this issue.

Thanks @TestAccessibilityGIT for all your help during this process.
